### PR TITLE
disable `hermes` for debug android builds & enable `hermes` for release builds

### LINKED ISSUE #18493: Send flow: "maxiumum call stack exceeded" error on the input_amount screen
The above error happens randomly when reaching the input_amount screen in debug mode.

Steps to reproduce:

Just going normally through the send flow: 
1. Enter address
2. Select token
3. Upon reaching the input_amount screen you may get that error

### PR DESCRIPTION
**Pull request by siddarthkay**
fixes #18493

## Summary

We enabled `hermes` for android in the `react-native` upgrade to `0.72.5` 
Although things seemed fine but developers were seeing frequent crashes in their local environment.

After some investigation the crashes were traced to max native call stack depth in `hermes` engine.
Disabling `hermes` for local debug builds helps fix that issue.

This PR disables `hermes` by default with the help of a exporting an environment variable in the `make run-android` command.
It is annoying that this also modifies `android/gradle.properties` so we keep `hermesEnabled` as `false` there as well.
We also enable `hermes` when generating release builds so that we can take advantage of `hermes` engine in release builds.

We also add a log to print whether `hermes` is enabled or not. I think its helpful to have this so that we know whether `hermes` is enabled or not.

## Testing notes
Test locally to see if max native call stack error goes away.

## Testing notes
no change in release builds, `hermes` was enabled in release builds since the upgrade was merged.
Testing is not required.

#### Platforms
- Android

status: ready

### CODE REVIEW

File: android/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## we generate this file at runtime from gradle.properties.local or gradle.properties.ci
+gradle.properties
```
Comment (jakubgs) [inline android/.gitignore]: Why is this necessary since it's already ignored in `.gitignore`?

File: nix/mobile/android/release.nix [hunk 2]
```diff
@@ -79,6 +79,8 @@ in stdenv.mkDerivation rec {
   STATUS_GO_SRC_OVERRIDE = statusGoSrcOverride;
   ANDROID_ABI_SPLIT = androidAbiSplit;
   ANDROID_ABI_INCLUDE = androidAbiInclude;
+  # hermes is Disabled by default, we enable it for release https://github.com/status-im/status-mobile/issues/18493
+  ORG_GRADLE_PROJECT_hermesEnabled = true;
```
Comment (jakubgs) [inline nix/mobile/android/release.nix]:
```suggestion
  # Disabled for debug builds to avoid 'maxiumum call stack exceeded' errors.
  # https://github.com/status-im/status-mobile/issues/18493
  ORG_GRADLE_PROJECT_hermesEnabled = true;
```

File: src/status_im/core.cljs [hunk 3]
```diff
@@ -34,8 +34,13 @@
 
 (def adjust-resize 16)
 
+(defn is-hermes
+  []
+  (boolean (.-HermesInternal js/global)))
```
Comment (jakubgs) [inline src/status_im/core.cljs:40]: Are those changes necessary?

File: Makefile [hunk 4]
```diff
@@ -273,6 +273,9 @@ run-re-frisk: ##@run Start re-frisk server
 
 # TODO: Migrate this to a Nix recipe, much the same way as nix/mobile/android/targets/release-android.nix
 run-android: export TARGET := android
+# INFO: We disable hermes for local builds by default to prevent crashes with exceeding native callstack errors
```
Comment (jakubgs) [inline Makefile]:
```suggestion
# Disabled for debug builds to avoid 'maxiumum call stack exceeded' errors.
```

File: android/gradle.properties [hunk 5]
```diff
@@ -57,4 +57,4 @@ newArchEnabled=false
 
 # Use this property to enable or disable the Hermes JS engine.
 # If set to false, you will be using JSC instead.
-hermesEnabled=true
+hermesEnabled=false
```
Comment (jakubgs) [inline android/gradle.properties:62]: No comment explaining why.

**Review by jakubgs — APPROVED** [review]

File: src/status_im/core.cljs (hunk 3 above)
Comment (siddarthkay) [reply src/status_im/core.cljs:40]: yeah this code helps to find out and prints if `hermes` is enabled or not
I think its useful to keep for future debugging sessions

File: src/status_im/core.cljs [hunk 6]
```diff
@@ -54,5 +59,6 @@
   (async-storage/get-item :screen-height #(reset! shell.state/screen-height %))
 
   (dev/setup)
+  (prn "hermesEnabled ->" (is-hermes))
```
Comment (ilmotta) [inline src/status_im/core.cljs]: @siddarthkay, at this point the logger is already set up, so it's better if we use it, e.g. `log/info`. Printing this info is useful. Good idea!

**Review by ilmotta — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im/core.cljs (hunk 3 above)
Comment (jakubgs) [reply src/status_im/core.cljs:40]: Cool.

**Review by jakubgs — APPROVED** [review]
Thanks for fixing this.
